lib/core/test_suite.js

```diff
@@ -171,6 +181,11 @@ export class TestSuite {
     this.#name = name;
   }
 
+  #initializeLocation(pathLocation) {
+    this.#ensureFilePathIsValid(pathLocation);
+    this.#location = new SuiteLocation(pathLocation);
```

Comment:
Let's move the validation inside `SuiteLocation` constructor, so that it's not possible to create `SuiteLocation` instances with a valid name